### PR TITLE
Add dataplane-operator docs

### PR DESCRIPTION
**Pull request by slagle**
Adds docs about the high level design of the dataplane-operator CRD's
and how they are used.

Also reorganizes the docs a bit and cleans up some of the README.md by
moving the contributing content to the dedicated contributing.md.

Signed-off-by: James Slagle <jslagle@redhat.com>

### CODE REVIEW

**Comment by slagle** [comment]
recheck

**Comment by slagle** [comment]
recheck

**Comment by slagle** [comment]
recheck

File: docs/contributing.md [hunk 1]
```diff
@@ -1,18 +1,161 @@
...
+
+```sh
+make undeploy
+```
+
+### How it works
+This project aims to follow the Kubernetes [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/)
+
+It uses [Controllers](https://kubernetes.io/docs/concepts/architecture/controller/)
+which provides a reconcile function responsible for synchronizing resources untile the desired state is reached on the cluster
```
Comment (fultonj) [inline docs/contributing.md:48]: - "provide" (singular) not "provides"
- until (not untile)

**Review by fultonj — COMMENTED** [review]
I'm basically +2 on this though I saw two grammar issues which should be easy to fix.

File: Makefile [hunk 2]
```diff
@@ -225,6 +228,16 @@ ginkgo: $(GINKGO) ## Download ginkgo locally if necessary.
 $(GINKGO): $(LOCALBIN)
 	test -s $(LOCALBIN)/ginkgo || GOBIN=$(LOCALBIN) go install github.com/onsi/ginkgo/v2/ginkgo
 
+.PHONY: kuttl-test
+kuttl-test: ## Run kuttl tests
```
Comment (fultonj) [inline Makefile:232]: This worked in my environment as documented with
`make kuttl-test KUTTL_ARGS="--test dataplane-deploy-no-nodes"`

File: docs/contributing.md (hunk 1 above)
Comment (slagle) [reply docs/contributing.md]: thanks, fixed these.

**Review by fultonj — APPROVED** [review]
+2

/lgtm

**Comment by slagle** [comment]
recheck

**Comment by slagle** [comment]
recheck

**Comment by fao89** [comment]
I think we can merge

**Comment by fao89** [comment]
recheck
/test dataplane-operator-build-deploy-kuttl
/lgtm

**Comment by fao89** [comment]
recheck

**Pull request merged**